pkg/kubelet/secret/secret_manager.go

```diff
@@ -201,8 +201,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 5 minutes (which is a lot, but...)
+	ttl := 5 * time.Minute
```

Comment:
@yujuhong - I didn't know that dynamic kubelet config is coming; this would be a good usecase for it.
As you suggested in the other PR (which I agree with), I will leave this PR open for now and we will see at the effects of #40360 before deciding what exactly to do with this PR.

@liggitt - I think that the volume-related things are slightly different. I think I agree with @yujuhong that we should try to address that separately.